docs/src/main/sphinx/client/client-protocol.md

```diff
@@ -0,0 +1,105 @@
+# Client protocol
+
+The Trino client protocol is a HTTP-based protocol that allows
+[clients](/client) to submit SQL queries and receive results.
+
+The protocol is a sequence of REST API calls to the
+[coordinator](trino-concept-coordinator) of the Trino
+[cluster](trino-concept-cluster). Following is a high-level overview:
+
+1. Client submits SQL query text to the coordinator of the Trino cluster.
+2. The coordinator starts processing the query.
+3. The coordinator returns a result set and a URI `nextUri` on the coordinator.
+4. The client receives the result set and initiates another request for more
+   data from the URI `nextUri`.
+5. The coordinator continues processing the query and returns further data with
+   a new URI.
+6. The client and coordinator continue with steps 4. and 5. until all
+   result set data is returned to the client or the clients stops requesting
+   more data.
+
+The client protocol supports two modes. Configure the [spooling
+protocol](protocol-spooling) for optimal performance for your clients.
+
+Detail spooling vs direct protocol here
+
+
+JSON compressed and other stuff
+
+pruning done by coordinator, only for stuff that wasnt already cleaned up by client
```

Comment:
correct. it's additional mechanism that allows to clean up the storage if either client doesn't acknowledge or the explicit-ack is disabled